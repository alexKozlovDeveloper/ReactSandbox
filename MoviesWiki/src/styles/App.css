h1 {
    color: #27aedb;
    text-align: center;
}

.body {
    background-color: dimgray;
}

.font {
    font-family: 'Graduate', cursive;
}