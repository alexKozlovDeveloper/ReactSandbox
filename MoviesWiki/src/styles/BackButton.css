.button {
    padding: 8px;
    font-size: 40px;
    background: url(https://static.thenounproject.com/png/95-200.png);
    min-width: 20px;
    min-height: 20px;
    background-size: contain;
    background-repeat: no-repeat;
    margin: 5px;
    cursor: pointer;
}