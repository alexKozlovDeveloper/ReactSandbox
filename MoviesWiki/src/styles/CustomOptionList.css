.container {
    height: 36px;
    padding: 5px;
}

.item {
    float: left;
    padding: 5px;
    margin: 2px;
    border: 1px solid #ccc;
    border-radius: 4px;
    font-family: cursive;
    cursor: pointer;
    background-color: #222327;
    color: white;
}

.isactive {
    background-color: gold;
}

.title {
    float: left;
    padding: 5px;
    margin: 2px;
    font-family: cursive;
    color: white;
}