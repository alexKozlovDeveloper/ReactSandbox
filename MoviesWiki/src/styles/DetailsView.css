.body {
    background-color: tomato;
    display: flex;
    padding-left: 30px;
    padding-right: 30px;
    padding-top: 10px;
    padding-bottom: 10px;
}

.imagecontainer {
    min-width: 150px;
}

.detailscontainer {
    padding: 10px;
}

.maintitlecontainer {
    display: flex;
}

.propertiescontainer {
    display: flex;
}

.descriptioncontainer {
    
}