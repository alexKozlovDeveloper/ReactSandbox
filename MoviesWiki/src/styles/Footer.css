.footer {
    bottom: 0;
    width: 100%;
    background: #333;
    padding: 10px 0;
    color: #fff;
    min-width: 500px;
}

.container {
    width: 80%;
    margin: 0 auto;
    text-align: center;
}