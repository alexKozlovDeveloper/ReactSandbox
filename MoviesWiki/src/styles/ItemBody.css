.body {
    background-color: #222327;
    padding: 4px;
}