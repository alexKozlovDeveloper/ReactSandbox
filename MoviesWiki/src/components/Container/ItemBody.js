import React, { Component } from "react";
import { connect } from 'react-redux'

import ItemTitle from "../Common/ItemTitle";
import ItemGenre from "../Common/ItemGenre";
import ItemImage from "../Common/ItemImage";
import ItemReleaseDate from "../Common/ItemReleaseDate";

import styles from "../../styles/ItemBody.css"

const updateSearchBy = (item) => ({
    type: 'UPDATE_SELECTED_ITEM',
    selectedItem: item
});

class ItemBody extends Component {
    constructor(props) {
        super(props);

        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(e) {
        debugger;
        this.props.updateSelectedItemFunc(this.props.item);
    }

    render() {        
        return (
            <div className={styles.body} onClick={this.handleClick}>
                <ItemImage image={this.props.item.poster_path} />
                <ItemReleaseDate releaseDate={this.props.item.release_date} />
                <ItemTitle title={this.props.item.title} />                
                <ItemGenre genre={this.props.item.genres.join(' ')} />            
            </div>
        );    
    }
}

function mapDispatchToProps(dispatch) {
    return {
         updateSelectedItemFunc: (item) => {
             dispatch(updateSearchBy(item))
         }       
    }
}

export default connect(null, mapDispatchToProps)(ItemBody);
