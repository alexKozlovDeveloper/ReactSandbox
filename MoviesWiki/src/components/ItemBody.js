import React, { Component } from "react";

import ItemTitle from "./ItemTitle";
import ItemGenre from "./ItemGenre";
import ItemImage from "./ItemImage";
import ItemReleaseDate from "./ItemReleaseDate";

import styles from "../styles/ItemBody.css"

class ItemBody extends Component {
    render() {
        return (
            <div className={styles.body}>
                <ItemTitle title={this.props.title} />
                <ItemGenre genre={this.props.genre} />
                <ItemImage image={this.props.image} />
                <ItemReleaseDate releaseDate={this.props.releaseDate} />
            </div>
        );
    }
}

export default ItemBody;