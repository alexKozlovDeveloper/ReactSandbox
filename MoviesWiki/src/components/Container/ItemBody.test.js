import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import rootReducer from '../../reducers'
import { createStore } from 'redux'
import { Provider } from 'react-redux'

import { BrowserRouter as Router, Link } from 'react-router-dom'

import ItemBody from "./ItemBody";
import styles from "../../styles/ItemBody.css"

configure({ adapter: new Adapter() });

test('ItemBody rendering test', () => {

    const store = createStore(rootReducer)

    const component = shallow(<Provider store={store}><ItemBody /></Provider>);
    expect(shallowToJson(component)).toMatchSnapshot();
});

test('ItemBody click test', () => {

    var item = {
        id: '99',
        title: "Interstellar 4",
        releaseDate: "2017",
        image: "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",
        genres: ["action 4"],
        rating: 7.2,
        duration: 145,
        description: "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
    }

    const mockCallBack = jest.fn();

    const store = createStore(rootReducer)

    const component = mount(<Router><Provider store={store}><ItemBody item={item} updateSelectedItemFunc={mockCallBack}/></Provider></Router>);

    component.findWhere(n => n.prop('to') === '/film/99').simulate('click');
    //component.find(Link).simulate('click');

    //expect(mockCallBack.mock.calls.length).toEqual(1);
});

