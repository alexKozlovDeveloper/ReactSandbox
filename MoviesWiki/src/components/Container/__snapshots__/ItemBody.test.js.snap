// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemBody rendering test 1`] = `
<div
  className="body"
>
  <ItemImage />
  <ItemReleaseDate />
  <ItemTitle />
  <ItemGenre />
</div>
`;
